distributed/deploy/tests/test_adaptive_core.py

```diff
@@ -89,3 +91,17 @@ async def test_interval():
     adapt._target = 10
     await asyncio.sleep(0.020)
     assert len(adapt.plan) == 1  # last value from before, unchanged
+
+
+@pytest.mark.asyncio
+@mock.patch("distributed.deploy.adaptive_core.AdaptiveCore.safe_target")
```

Comment:
> Nitpick: Instead of using mock could we instead create an explicit adaptive subclass (similar to MyAdaptive) which raises an OSError when attempting to adapt (see the diff below)?

I like this idea. I too find mocking confusing sometimes and hard to reason about for users who didn't write the mocking code... I will update the test shortly